### DISCUSSION TITLE
Audio Kit v2.2 + IMU MPU9250 - I2C Error

### DISCUSSION THREAD

**Discussion by FrancescoGanis** [pschatzmann/arduino-audiokit#98] [Q&A] [answered]
Hi, first of all I would like to thank the developer for this amazing library!!

GENERAL INFO
- Board: V2.2 2748 with an AC101 (I assume)
- IMU: MPU9250 
- Arduino IDE

CONNECTION IMU-Board: 
- Wire, I2C
- SDA board pin = 21
- SCL board pin = 22
- IMU bus = 0x68
- IMU clock = 100000

PROJECT DESCRIPTION
I just started working with it and I would like to play a sound stored in the SD card mounted in the board when the accelerometer of the IMU detects a movement on the Z axes. 

- Read and play the audio file: I used the provided example for reading and reproducing the .mp3 sound file on the SD and it works like a charm. 
- Read and set a flag from the IMU: I adapted an example script from the library Bolder Flight Systems MPU9250 to read the accelerometer data and set a flag when the Z axes output passes a certain threshold. 

ERROR
When I merge the two snippets of code I get the following error in the Serial Monitor: 
**E (2500) i2c: i2c driver install error**
I tried to use different GPIO for the I2C connection between IMU and the board and nothing changed. 

I am sure I'm missing something so I appreciate any help or hints! 

Thanks

```
#include "Arduino.h"
#include "mpu9250.h"
#include "AudioTools.h"
#include "AudioLibs/AudioKit.h"
#include "AudioLibs/AudioSourceSD.h" // or AudioSourceIdxSD.h
#include "AudioCodecs/CodecMP3Helix.h"

/* Mpu9250 object */
bfs::Mpu9250 imu;

// Variables for tracking box movement
bool isBoxPickedUp = false;
const unsigned short pickUpThreshold = 10;

// I2C addresses for IMU
#define I2C_SDA 21 
#define I2C_SCL 22 

// Timer variables for playing sound
unsigned long currentTime = millis();
unsigned long lastSoundTime = 0;
const unsigned long soundInterval = 3000; // 3 seconds in milliseconds

// Audio player variables
const char *startFilePath="/";
const char* ext="mp3";
AudioSourceSD source(startFilePath, ext, PIN_AUDIO_KIT_SD_CARD_CS);
AudioKitStream kit;
MP3DecoderHelix decoder;  // or change to MP3DecoderMAD
AudioPlayer player(source, kit, decoder);

void Connect_IMU(){

  // // Start the I2C bus
  Wire.begin(I2C_SDA, I2C_SCL);
  Wire.setClock(100000); // Original from example 400000

  // // I2C bus,  0x68 address
  imu.Config(&Wire, bfs::Mpu9250::I2C_ADDR_PRIM);

  // Initialize and configure IMU
  if (!imu.Begin()) {
    Serial.println("Error initializing communication with IMU");
    while(1) {}
  }

  // Set the sample rate divider
  if (!imu.ConfigSrd(19)) {
    Serial.println("Error configured SRD");
    while(1) {}
  }
}

void setup() {
 // serial to display data
  Serial.begin(115200);

  while (!Serial && millis() < 5000);

  delay(500);

  // =========== AUDIO PLAYER ===========
  AudioLogger::instance().begin(Serial, AudioLogger::Info);

  // setup output
  auto cfg = kit.defaultConfig(TX_MODE);
  // sd_active is setting up SPI with the right SD pins by calling 
  // SPI.begin(PIN_AUDIO_KIT_SD_CARD_CLK, PIN_AUDIO_KIT_SD_CARD_MISO, PIN_AUDIO_KIT_SD_CARD_MOSI, PIN_AUDIO_KIT_SD_CARD_CS);
  cfg.sd_active = true;
  kit.begin(cfg);
  cfg.pins.i2c_scl = 32;
  cfg.pins.i2c_sda = 33;

  // setup player
  player.setVolume(0.7);
  player.begin();

  // select file with setPath() or setIndex()
  // player.setPath("victory.mp3");
  player.setIndex(0); // 2nd file

  // =========== IMU ===========
  // Connect to IMU with I2C
  Connect_IMU();

}

void loop() {
  // Read IMU accelerometer data
  // if (imu.Read()) {
  //   Serial.print("Acc X: ");
  //   Serial.print(imu.accel_x_mps2());
  //   Serial.print("\t");
  //   Serial.print("Acc Y: ");
  //   Serial.print(imu.accel_y_mps2());
  //   Serial.print("\t");
  //   Serial.print("Acc Z: ");
  //   Serial.print(imu.accel_z_mps2());
  //   Serial.print("\n");
  //  }

    // If the box is picked up
    if (imu.Read() && imu.accel_z_mps2() > pickUpThreshold)
    {

      currentTime = millis(); // Update current time

      // if 3 or more seconds have passed before last sound
      if ((currentTime - lastSoundTime >= soundInterval)) {
        isBoxPickedUp = true;
        lastSoundTime = millis();
        Serial.print("Play sound");
        Serial.print("\n");

        player.copy();
      }
    }
    else {
      isBoxPickedUp = false;
    }
}
```

**Comment by pschatzmann** [comment, marked as answer]
Did you already read the corresponding documentation that can be found in the Wiki?
e.g.
- https://github.com/pschatzmann/arduino-audiokit/wiki/SPI-and-I2C

**Reply by FrancescoGanis** [reply]
Yes I did, thank you. 
Do you suggest to use the ESP32 I2C API based implementation instead of wire library? You are using Wire1 instead of Wire to let the user to freely work with it but still I have this issue.

**Comment by pschatzmann** [comment]
I think you rather have a pin conflict. Did you double check the used pins: https://github.com/pschatzmann/arduino-audiokit/blob/main/src/audio_board/ai_thinker_ac101.h

#define PA_ENABLE_GPIO 21
This is the power amplifier!

**Reply by FrancescoGanis** [reply]
Yes that is what I guessed too. I changed the IMU with other ones that are available on the board but I got the same error. 
I also saw from the _ai_thinker_ac101.h_ file that the pins not assigned are 3, 4, 6-12, 16, 17, 20, 24, 28-31 but none of them are available on the pinout of the board.

**Reply by pschatzmann** [reply]
I never used these pins, so I can't give any advise. 
I just remember that others were failing using them as well...

**Reply by FrancescoGanis** [reply]
Thank you, I will try to investigate further into it.
